dvc/stage/run.py

```diff
@@ -70,32 +70,33 @@ def cmd_run(stage, *args, checkpoint_func=None, **kwargs):
         kwargs["shell"] = False
         executable = os.getenv("SHELL") or "/bin/sh"
         warn_if_fish(executable)
-        cmd = _nix_cmd(executable, stage.cmd)
+        cmd = [_nix_cmd(executable, _cmd) for _cmd in cmd]
 
     main_thread = isinstance(
         threading.current_thread(),
         threading._MainThread,  # pylint: disable=protected-access
     )
-    old_handler = None
-    p = None
+    for _cmd in cmd:
+        old_handler = None
+        p = None
 
-    try:
-        p = subprocess.Popen(cmd, **kwargs)
-        if main_thread:
-            old_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
-
-        killed = threading.Event()
-        with checkpoint_monitor(stage, checkpoint_func, p, killed):
-            p.communicate()
-    finally:
-        if old_handler:
-            signal.signal(signal.SIGINT, old_handler)
-
-    retcode = None if not p else p.returncode
-    if retcode != 0:
-        if killed.is_set():
-            raise CheckpointKilledError(stage.cmd, retcode)
-        raise StageCmdFailedError(stage.cmd, retcode)
+        try:
+            p = subprocess.Popen(_cmd, **kwargs)
+            if main_thread:
+                old_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+            killed = threading.Event()
+            with checkpoint_monitor(stage, checkpoint_func, p, killed):
+                p.communicate()
+        finally:
+            if old_handler:
+                signal.signal(signal.SIGINT, old_handler)
+
+        retcode = None if not p else p.returncode
+        if retcode != 0:
+            if killed.is_set():
+                raise CheckpointKilledError(stage.cmd, retcode)
+            raise StageCmdFailedError(stage.cmd, retcode)
```

Comment:
This will result in the whole list of commands being added to the exception. Feels like it should be a bit more granular in case of a list?

Reply:
Also, in `run_stage`, there is a logger.info message about running the commands and right now it will show just a plain single-line list of commands:
```
Running stage 'foo' with command:          
        ['echo foo > foo', 'exit 1', 'echo bar > bar']
```
maybe let's adjust it for the list?

Reply:
I have moved the command logging to the `cmd_run` function and updated the `_cmd` arg of the errors